### PR TITLE
add stage builds in preparation for ci pipeline and e2e tests

### PR DESCRIPTION
**Pull request by grohmio**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -67,7 +67,42 @@
...
+                  "type": "anyComponentStyle",
+                  "maximumWarning": "6kb",
+                  "maximumError": "10kb"
+                }
+              ]
+            },
+            "de": {
+              "localize": [
+                "de"
+              ]
```
Comment (christopher-l) [inline angular.json:105]: duplicate
```suggestion
```

File: angular.json [hunk 2]
```diff
@@ -151,5 +186,8 @@
       }
     }
   },
-  "defaultProject": "open-edu-hub-frontend"
-}
\ No newline at end of file
+  "defaultProject": "open-edu-hub-frontend",
+  "cli": {
+    "analytics": "d8fda401-28c0-4466-a895-88ee251efe98"
+  }
```
Comment (christopher-l) [inline angular.json:192]: What does this do?

File: package.json [hunk 3]
```diff
@@ -7,6 +7,8 @@
     "clean": "rm -rf dist",
     "prebuild": "graphql-codegen",
     "build": "ng build --prod --localize",
+    "build-dev": "ng build --localize",
+    "build-stage": "ng build --configuration=staging --localize",
```
Comment (christopher-l) [inline package.json:11]: Please take care to apply the `prebuild` script also for the new `build-*` jobs.

File: src/environments/environment.stage.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const environment = {
+    production: true,
+    relayUrl: 'https://staging.wirlernenonline.de/relay/graphql',
+    editorBackendUrl: 'https://staging.wirlernenonline.de/editor/graphql',
```
Comment (christopher-l) [inline src/environments/environment.stage.ts:4]: The format has slightly changed, not including the `/graphql` anymore.
```suggestion
    relayUrl: 'https://staging.wirlernenonline.de/relay',
    editorBackendUrl: 'https://staging.wirlernenonline.de/editor',
```

File: angular.json (hunk 2 above)
Comment (grohmio) [reply angular.json:192]: just a copy from production

File: angular.json (hunk 1 above)
Comment (grohmio) [reply angular.json:105]: in staging we need the fixed url 'https://staging.wirlernenonline.de/relay/graphql' this is used by the pipeline later on, so staging in this context means pipeline installations

production should use the dynamic approach window.location.origin + '/relay',

File: package.json (hunk 3 above)
Comment (grohmio) [reply package.json]: done

File: src/environments/environment.stage.ts (hunk 4 above)
Comment (grohmio) [reply src/environments/environment.stage.ts]: done

File: angular.json (hunk 1 above)
Comment (grohmio) [reply angular.json]: rest done

**Pull request merged**